Python/compile.c

```diff
@@ -2544,7 +2544,16 @@ compiler_class_body(struct compiler *c, stmt_ty s, int firstlineno)
         return ERROR;
     }
     assert(c->u->u_static_attributes);
-    PyObject *static_attributes = PySequence_Tuple(c->u->u_static_attributes);
+    PyObject *static_attributes_unsorted = PySequence_List(c->u->u_static_attributes);
+    if (static_attributes_unsorted == NULL) {
+        return ERROR;
```

Comment:
Shouldn't these early exits call `compiler_exit_scope(c)` like the existing early exit does?

Reply:
```suggestion
        compiler_exit_scope(c);
        return ERROR;
```

Reply:
No change needed on main; the code is quite different there. Sorry for missing this while creating the backport.